File: content/snippets/go/language_basics_nullable_optional_values.go
package main

import "fmt"

func parseAge(text string) *int {
	if text == "" {
		return nil
	}

	age := 29
	return &age
}

func main() {
	age := parseAge("29")
	missing := parseAge("")

	fmt.Println("age present:", age != nil)
	if missing == nil {
		fmt.Println("missing default:", 0)
	} else {
		fmt.Println("missing default:", *missing)
	}
}

File: content/snippets/cpp/language_basics_nullable_optional_values.cpp
#include <iostream>
#include <optional>
#include <string>

std::optional<int> parse_age(const std::string& text) {
    if (text.empty()) {
        return std::nullopt;
    }
    return std::stoi(text);
}

int main() {
    auto age = parse_age("29");
    auto missing = parse_age("");

    std::cout << "age present: " << age.has_value() << "\n";
    std::cout << "missing default: " << missing.value_or(0) << "\n";
    return 0;
}
